math/krishnamurthy.c

```diff
@@ -0,0 +1,76 @@
+#include <stdio.h> // For I/O ops
+#include <assert.h> //For test cases
+
+/**
+SCRIPT DESCRIPTION:
+
+ * @description a krishnamurthy number is
+ * a number whose sum of the factorial of digits
+ * is equal to the number itself.
+
+ * @example 145 = 1! + 4! + 5!
+
+ * @reference https://www.geeksforgeeks.org/problems/krishnamurthy-number1323/1
+*/
+
+
+/**
+ * @description n! = n*(n-1)*(n-2)*...*1
+ * @brief calculates the factorial of a number
+ * @param n the number we want to calculate the factorial
+ * @return an integer representing the factorial of n
+ */
+int factorial(int n){
```

Comment:
there are implementations of factorial available in this repository you could use that if you'd like

Reply:
i didn't find any satisfying implementation of the factorial, so i made my own! The implementations already established were not implemented using a functional approach!!!